src/pages/Admin/Venue/Details/VenueDetails.tsx

```diff
@@ -0,0 +1,254 @@
+
+// Hooks
+import { Link } from "react-router-dom";
+
+// Typings
+import { VenueDetailsProps } from "./VenueDetails.types";
+
+// Styles
+import * as S from "./VenueDetails.styles";
+import MapPreview from "pages/Admin/MapPreview";
+import { updateRoom } from "api/admin";
+import { RoomData_v2 } from "types/RoomData";
+import { useFirestoreConnect } from "react-redux-firebase";
+import { isEqual } from "lodash";
+import RoomDeleteModal from "../Rooms/RoomDeleteModal";
+import { VenueOwnersModal } from "pages/Admin/VenueOwnersModal";
+import { useUser } from "hooks/useUser";
+import { AnyRoom } from "types/Venue";
+
+type Owner = {
+  id: string;
+  data: unknown;
+  partyName: string;
+  pictureUrl: string;
+};
+
+type EditRoomType = RoomData_v2 & {
+  roomIndex: number;
+};
+
+const VenueDetails: React.FC<VenueDetailsProps> = ({ venue }) => {
+  const { name, owners, id: venueId, rooms, mapBackgroundImageUrl } = venue;
+  const {
+    subtitle,
+    description,
+    coverImageUrl,
+  } = venue.config.landingPageConfig;
+  const { icon } = venue.host;
+  const { user } = useUser();
+
+  const [ownersData, setOwnersData] = useState<Owner[]>([]);
+  const [modalOpen, setModalOpen] = useState<boolean>(false);
+
+  const [editingRoom, setEditingRoom] = useState<EditRoomType | null>(null);
+  const [showEventModal, setShowEventModal] = useState(false);
+  const [eventRoom, setEventRoom] = useState("");
+  const [showDeleteModal, setShowDeleteModal] = useState(false);
+  const [showOwnersModal, setShowOwnersModal] = useState(false);
+
+  useFirestoreConnect([
+    {
+      collection: "venues",
+      doc: venueId,
+      subcollections: [{ collection: "events" }],
+      orderBy: ["start_utc_seconds", "asc"],
+      storeAs: "events",
+    },
+  ]);
+
+  const ownersRef = useRef([]);
+
+  useEffect(() => {
+    const newOwners: Owner[] = [];
+    async function getOwnersData() {
+      if (owners && owners.length > 0) {
+        for (const owner of owners) {
+          const user = await firebase
+            .functions()
+            .httpsCallable("venue-getOwnerData")({ userId: owner });
+
+          const userData = user.data;
+
+          if (ownersRef.current.filter((i: Owner) => i.id !== owner)) {
+            newOwners.push({
+              id: owner,
+              ...userData,
+            });
+          }
+        }
+
+        if (!isEqual(ownersData, newOwners)) {
+          setOwnersData(newOwners);
+        }
+      }
+    }
+    getOwnersData();
+  }, [owners, ownersData]);
+
+  if (!user) return null;
+
+  const toggleRoomModal = () => setModalOpen(!modalOpen);
+
+  const handleEditRoom = (room: RoomData_v2, index: number) => {
+    setEditingRoom({
+      ...room,
+      roomIndex: index,
+    });
+  };
+
+  const handleRoomEvent = (roomName: string) => {
+    setEventRoom(roomName);
+    setShowEventModal(true);
+  };
+
+  const handleEditRoomSave = (values: RoomData_v2, index: number) => {
```

Comment:
.

Reply:
Also probably use semantic names for these functions. I haven't deeply looked at what is being done here, but `handleEditRoomSave` could probably just be `saveRoom` or similar.